lib/frames.rb

```diff
@@ -0,0 +1,77 @@
+require './lib/score_card'
+require './lib/roll'
+
+class Frames
+
+  def initialize
+    @round = 1
+    @score_count = 0
+    @strike_count = 0
+    @bonus = nil
+    @first_roll = true
+    @this_frame_count = 0
+  end
+
+  def pins_knocked
+    puts "How many pins did you knock down"
+    pins = gets.chomp.to_i
+  end
+
+
+  def new_round
+    if @round >= 11
+      ScoreCard.new(@score_count, @strike_count)
+    else
+      new_roll
+    end
+  end
+
+  def new_roll
+    @roll = Roll.new(score_count: @score_count, strike_count: @strike_count, first_roll: @first_roll, this_frame_count: @this_frame_count)
```

Comment:
Each roll currently needs to know a lot about what's going on in the rest of the game, which suggests that maybe some responsibilities could be moved out of it, or the class should be renamed. Ideally, a roll should be pretty independent of the rest of the game.